jib-core/src/main/java/com/google/cloud/tools/jib/builder/BuildConfiguration.java

```diff
@@ -178,6 +188,69 @@ public BuildConfiguration build() {
           throw new IllegalStateException(errorMessage.toString());
       }
     }
+
+    /**
+     * Converts/validates a list of ports with ranges to an expanded form without ranges.
+     *
+     * <p>Example: ["1000/tcp", "2000-2002/tcp"] -> ["1000/tcp", "2000/tcp", "2001/tcp", "2002/tcp"]
+     *
+     * @param ports the list of port numbers/ranges
+     * @return the ports as a list of integers
+     * @throws NumberFormatException if any of the ports are in an invalid format or out of range
+     */
+    @VisibleForTesting
+    ImmutableList<String> expandPortRanges(List<String> ports) throws NumberFormatException {
+      ImmutableList.Builder<String> result = new ImmutableList.Builder<>();
+
+      for (String port : ports) {
+        // Make sure configuration is a single number or a range, with an optional protocol
+        // Examples:
+        //   100
+        //   200-210
+        //   1000/tcp
+        //   2000/udp
+        //   500-600/tcp
+        if (!port.matches("^\\d+(-\\d+)?(\\/tcp|\\/udp)?$")) {
+          throw new NumberFormatException(
+              "Invalid port configuration: '"
+                  + port
+                  + "'. Make sure the port is a single number or a range of two numbers separated "
+                  + "with a '-', with or without protocol specified (e.g. '<portNum>/tcp' or "
+                  + "'<portNum>/udp').");
+        }
+
+        // Parse protocol
+        List<String> splitProtocol = Splitter.on("/").splitToList(port);
+        String protocol = splitProtocol.size() > 1 ? "/" + splitProtocol.get(1) : "";
+
+        // Parse range (or treat as range of min-min if only single port configuration)
+        List<String> splitRange = Splitter.on("-").splitToList(splitProtocol.get(0));
```

Comment:
Since you are using pattern match above already, it might make sense to just use the regex matcher to get the parts, so the regex would be like:

```
(\d+)(-\d+|)(/tcp|/udp|)
```

So that it would match 4 groups, where 1 is min, 2 is max or nothing, and 3 is protocol or nothing.